src/components/content-region/index.jsx

```diff
@@ -58,12 +59,21 @@ function SiblingNav({ route, lang, start }) {
 export default function ContentRegion({ content, components, ...props }) {
 	const hasNav = !!(props.next || props.prev);
 	components = Object.assign({}, COMPONENTS, components);
+
+	useEffect(() => {
+		const hash = location.hash;
+		if (hash) {
+			// Hack to force a scroll
+			location.hash = '';
+			location.hash = hash;
+		}
+	}, [props.current]);
+
```

Comment:
This still isn't perfect but works more often anyhow.

Ideas certainly welcome if ya got 'em.